packages/core/lib/debug/printer.js

```diff
@@ -209,10 +210,33 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    // printout instructions
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentIndex = instruction.index;
+    // printout 3 previous instructions
```

Comment:
A very minor nit, I think this comment should explain the the logic for printing the ellipse.

Reply:
add a comment:
```
// add an ellipse if the first instruction of the stack will not be included in the printout
```

Reply:
That's not the clearest, but it'll do, I suppose.  (Instructions don't go on the stack, and it would make more sense to emphasize that there *exist* additional instructions before, rather than that the first one *isn't* present, but whatever, ultimately it gets the point across.)

Reply:
(Sorry, normally I wouldn't nitpick comments like this!  Just it seems like you could use some help with your prose writing, so I thought I'd point out how it could be written clearer. :) )

Reply:
changes made